### DISCUSSION TITLE
VLAN Groups one device

### DISCUSSION THREAD

**Discussion by jscooksey** [netbox-community/netbox#8571] [Q&A] [answered]
VLAN Groups gives us a nice list showing available VLANs within that group
I have a need to not allow the same VLAN number in different groups though, which is allowed
I use a range of say 1000-2000 for one purpose, then 2000-3000 for another.
I'd like to see whats available in each range when I go to use it, but the VLAN Group allow the same vlan ID to be in different groups
Is there a way to do this?

**Comment by candlerb** [comment, marked as answer]
I think you'd have to use something like a [custom validator](https://netbox.readthedocs.io/en/stable/customization/custom-validation/) with python code that does a database lookup.

VLAN IDs are necessarily reusable.  You might have a VLAN 100 in one site and different VLAN 100 in another site. Indeed, you might have two distinct VLANs, both with ID 100, in the same site.  So there is no VID uniqueness enforcement in Netbox.
